### PR TITLE
Fix 'wasPressed' never returning true

### PR DESCRIPTION
**Pull request by OmarShehata**
As detailed in #984 , `GamePads.wasPressed()` will never return true.

This happens because `update()` sets `previous` to `current` before updating `current`, but they're all pointing to the same object! 

A potential fix is to make sure to only copy the values. For the convenience of the reviewer I also added a gamepad test to confirm that it currently does not work, and will work with this fix. 

This version is a lot more verbose than the previous, but it works. I welcome suggestions on making it more concise/clear!

### CODE REVIEW

**Comment by willeastcott** [comment]
@daredevildave, can you review please?

**Comment by OmarShehata** [comment]
Any updates/feedback @daredevildave ?

**Comment by daredevildave** [comment]
Hi Omar,

Can you complete the Contributors License Agreement? 

https://docs.google.com/forms/d/e/1FAIpQLSfG9XHIgdapCBkvwxbXow592Y5NFPEKD5TM-TyWOmbaCbLeHA/viewform

Thanks

File: src/input/game-pads.js [hunk 1]
```diff
@@ -107,9 +107,31 @@ pc.extend(pc, function () {
 
             var i, len = pads.length;
             for (i = 0;i < len; i++) {
-                this.previous[i] = this.current[i];
-                this.current[i] = pads[i];
-            }            
+                // Initialize the previous and current for each gamepad
+                if(this.previous[i] == null){
```
Comment (daredevildave) [inline src/input/game-pads.js]: Please use === always 
And spacing of `if` statements must fit the coding standards doc: https://github.com/playcanvas/engine/blob/master/CONTRIB.md

**Comment by daredevildave** [comment]
Just tested and it seems to work. Needs to be fixed to [coding standards](https://github.com/playcanvas/engine/blob/master/CONTRIB.md) and CLA completed then I'll merge. 

Thanks!

**Comment by OmarShehata** [comment]
@daredevildave I've completed the CLA and fixed all the style issues, except for making each variable be declared on an individual line, because I noticed other parts in the code use this pattern:

```
var i, len = padDevices.length;
for (i = 0; i < len; i++) {
```

I've also made the gamepad example have an example of `wasPressed` as well as `isPressed`. That file isn't required for this PR but might be nice to have.

File: src/input/game-pads.js [hunk 2]
```diff
@@ -104,12 +104,33 @@ pc.extend(pc, function () {
...
-            var i, len = pads.length;
-            for (i = 0;i < len; i++) {
-                this.previous[i] = this.current[i];
-                this.current[i] = pads[i];
-            }            
+            var i, j, len = pads.length;
+            var buttonsLen;
+            for (i = 0; i < len; i++) {
+                // Initialize the previous and current for each gamepad
+                if(this.previous[i] == null) {
```
Comment (daredevildave) [inline src/input/game-pads.js]: Please use === always

File: src/input/game-pads.js [hunk 3]
```diff
@@ -104,12 +104,33 @@ pc.extend(pc, function () {
...
+                    this.previous[i] = { pad: { buttons: [ ] }, map:{ } };
+                    this.current[i] = { pad: { buttons: [ ] }, map:{ } };
+                }
+                // Copy values to previous from current
+                buttonsLen = this.current[i].pad.buttons.length;
+                for(j = 0; j < buttonsLen; j++) {
+                    if(this.previous[i].pad.buttons[j] == undefined) {
```
Comment (daredevildave) [inline src/input/game-pads.js]: ===

File: src/input/game-pads.js [hunk 4]
```diff
@@ -104,12 +104,33 @@ pc.extend(pc, function () {
...
+                        this.previous[i].pad.buttons[j] = { pressed: false };
+                    }
+                    this.previous[i].pad.buttons[j].pressed = this.current[i].pad.buttons[j].pressed;
+                }
+                // Copy values to current from pads
+                buttonsLen = pads[i].pad.buttons.length;
+                for(j = 0; j < buttonsLen; j++) {
+                    if(this.current[i].pad.buttons[j] == undefined) {
```
Comment (daredevildave) [inline src/input/game-pads.js]: ===

**Comment by daredevildave** [comment]
Change == to === and I think we're good.

**Comment by OmarShehata** [comment]
Ah, yes, sorry about that! I started going through the style guide point by point and forgot about strict equality. 

Should be good now.

**Comment by willeastcott** [comment]
Thanks! 👍

**Pull request merged**